core/citrus-base/src/main/java/org/citrusframework/TestCaseRunnerFactory.java

```diff
@@ -0,0 +1,86 @@
+package org.citrusframework;
+
+import org.citrusframework.context.TestContext;
+import org.citrusframework.exceptions.CitrusRuntimeException;
+import org.citrusframework.spi.ResourcePathTypeResolver;
+
+/**
+ * Factory for creating {@link TestCaseRunner} instances. By default, it uses
+ * Citrus' built-in runner, but it also offers the flexibility to replace the default runner with a
+ * custom implementation. To do this, it leverages the Citrus {@link ResourcePathTypeResolver}
+ * mechanism.
+ *
+ * To provide a custom runner, the following file needs to be added to the classpath:
+ * <p>
+ * <code>
+ * 'META-INF/citrus/test/runner/custom'
+ * </code>
+ * </p>
+ * The specified file must define the type of {@link TestCaseRunnerProvider} responsible for
+ * delivering the custom test case runner.
+ *
+ * @author Thorsten Schlathoelter
+ * @since 4.0
+ * @see TestCaseRunnerProvider
+ */
+public class TestCaseRunnerFactory {
+
+
+    /** The key for the default Citrus test case runner provider */
+    private static final String DEFAULT = "default";
+
+    /** The key for a custom test case runner provider */
+    private static final String CUSTOM = "custom";
+
+    /** Test runner resource lookup path */
+    private static final String RESOURCE_PATH = "META-INF/citrus/test/runner";
+
+    /** Default Citrus test runner from classpath resource properties. Non-final to support testing.*/
+    private ResourcePathTypeResolver typeResolver = new ResourcePathTypeResolver(RESOURCE_PATH);
```

Comment:
since you're using `ReflectionTestUtils` anyway, I don't think the `final` keyword would make a difference. right?